### PR TITLE
Refactor/heat

### PR DESCRIPTION
**Pull request by Jyrsa**
I can't think of anything other than some individual configs missing in this.

Everything seems to work (see internal configs for all the other pieces). Updating environment variables requires re-building the stateful containers which is a 10-20 minute wait which kind of sucks. We should prioritize building stateless docker images like we want them in the next stage.

### CODE REVIEW

File: roles/common/tasks/fetch_application.yml [hunk 1]
```diff
@@ -19,11 +19,19 @@
...
-- name: Install latest version of Sphinx
-  pip: name=Sphinx state=latest virtualenv={{ virtualenv_path }}
+# some packages fail on some versions of pip when installing alongside sphinx
+# because they don't declare a dependency
+# positional fails if pbr isn't installed
+
+- name: Install latest version of separately installable software
+  pip: name={{ item }} state=latest virtualenv={{ virtualenv_path }}
+  with_items:
+    - Sphinx
```
Comment (apurva3000) [inline roles/common/tasks/fetch_application.yml]: sphinx again? Weren't we supposed to get rid of it

File: roles/single_server_with_docker/tasks/main.yml [hunk 2]
```diff
@@ -158,41 +165,58 @@
...
+    state: directory
+    suffix: own
+  register: img_build
+
 - name: Pull images
-  docker_image: path={{ mktemp.stdout }} name={{ item }} state=present docker_host=unix:///var/run/docker.sock
+  docker_image:
+    name: "{{ item }}"
+    state: present
+    docker_host: unix:///var/run/docker.sock
```
Comment (apurva3000) [inline roles/single_server_with_docker/tasks/main.yml]: I think this (line 214) can be removed altogether

File: roles/single_server_with_docker/tasks/main.yml [hunk 3]
```diff
@@ -229,11 +253,28 @@
     state: started
     restart_policy: always
 
+- name: combine common and separate environment variables
```
Comment (apurva3000) [inline roles/single_server_with_docker/tasks/main.yml]: 👍

**Comment by Jyrsa** [comment]
@Ilavarasi1 okayed this before going on holiday. Going to rebase and merge a squashed version to keep version history neater.

**Pull request merged**